README.md
# Scala Exercises - Stdlib library

This repository hosts a content library for the [Scala Exercises](https://www.scala-exercises.org/) platform, including interactive exercises related to the Scala standard library.

## About Scala Exercises

"Scala Exercises" brings exercises for the Stdlib, Cats, Shapeless, and many other great libraries for Scala to your browser. It offers hundreds of solvable exercises organized into several categories covering the basics of the Scala language and its most important libraries.

Scala Exercises is available at [scala-exercises.org](https://scala-exercises.org).

## Contributing

Contributions welcome! Please join our [Gitter channel](https://gitter.im/scala-exercises/scala-exercises)
to get involved, or visit our [GitHub site](https://github.com/scala-exercises).

## License

Copyright (C) 2015-2019 47 Degrees, LLC.
Reactive, scalable software solutions.
http://47deg.com
hello@47deg.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


CHANGELOG.md
# Changelog

## [v0.6.6](https://github.com/scala-exercises/exercises-stdlib/tree/v0.6.6) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-stdlib/compare/v0.6.5...v0.6.6)

📈 **Dependency updates**

- Update definitions, exercise-compiler, ... to 0.6.4 [\#218](https://github.com/scala-exercises/exercises-stdlib/pull/218) ([47erbot](https://github.com/47erbot))

## [v0.6.5](https://github.com/scala-exercises/exercises-stdlib/tree/v0.6.5) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-stdlib/compare/v0.6.4...v0.6.5)

📈 **Dependency updates**

- Update sbt-mdoc to 2.2.4 [\#217](https://github.com/scala-exercises/exercises-stdlib/pull/217) ([47erbot](https://github.com/47erbot))
- Update scalacheck-1-14 to 3.2.1.0 [\#216](https://github.com/scala-exercises/exercises-stdlib/pull/216) ([47erbot](https://github.com/47erbot))
- Update scalatest to 3.2.1 [\#215](https://github.com/scala-exercises/exercises-stdlib/pull/215) ([47erbot](https://github.com/47erbot))
- Update sbt-scalafmt to 2.4.2 [\#214](https://github.com/scala-exercises/exercises-stdlib/pull/214) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.4 [\#213](https://github.com/scala-exercises/exercises-stdlib/pull/213) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.3 [\#212](https://github.com/scala-exercises/exercises-stdlib/pull/212) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.2 [\#211](https://github.com/scala-exercises/exercises-stdlib/pull/211) ([scala-steward](https://github.com/scala-steward))

## [v0.6.4](https://github.com/scala-exercises/exercises-stdlib/tree/v0.6.4) (2020-06-30)

[Full Changelog](https://github.com/scala-exercises/exercises-stdlib/compare/v0.6.3...v0.6.4)

## [v0.6.3](https://github.com/scala-exercises/exercises-stdlib/tree/v0.6.3) (2020-06-29)

[Full Changelog](https://github.com/scala-exercises/exercises-stdlib/compare/v0.6.2...v0.6.3)

📈 **Dependency updates**

- Update sbt to 1.3.13 [\#208](https://github.com/scala-exercises/exercises-stdlib/pull/208) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.3 [\#207](https://github.com/scala-exercises/exercises-stdlib/pull/207) ([scala-steward](https://github.com/scala-steward))
- Update scalacheck-1-14 to 3.2.0.0 [\#204](https://github.com/scala-exercises/exercises-stdlib/pull/204) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.2.0 [\#203](https://github.com/scala-exercises/exercises-stdlib/pull/203) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.2 [\#201](https://github.com/scala-exercises/exercises-stdlib/pull/201) ([scala-steward](https://github.com/scala-steward))
- Update sbt to 1.3.11 [\#197](https://github.com/scala-exercises/exercises-stdlib/pull/197) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.1.2 [\#179](https://github.com/scala-exercises/exercises-stdlib/pull/179) ([scala-steward](https://github.com/scala-steward))

**Merged pull requests:**

- Update definitions, exercise-compiler, ... to 0.6.3 [\#209](https://github.com/scala-exercises/exercises-stdlib/pull/209) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.1 [\#205](https://github.com/scala-exercises/exercises-stdlib/pull/205) ([BenFradet](https://github.com/BenFradet))
- Prepare repository for next  release and SBT build improvements [\#200](https://github.com/scala-exercises/exercises-stdlib/pull/200) ([juanpedromoreno](https://github.com/juanpedromoreno))
- Update Classes.scala [\#198](https://github.com/scala-exercises/exercises-stdlib/pull/198) ([Czajnikowski](https://github.com/Czajnikowski))
- Update Enumerations.scala [\#190](https://github.com/scala-exercises/exercises-stdlib/pull/190) ([joelittlejohn](https://github.com/joelittlejohn))
- Tidy comments in ByNameParameter.scala [\#189](https://github.com/scala-exercises/exercises-stdlib/pull/189) ([joelittlejohn](https://github.com/joelittlejohn))
- unapplySeq is useful when number of subvalues ISN'T fixed [\#188](https://github.com/scala-exercises/exercises-stdlib/pull/188) ([joelittlejohn](https://github.com/joelittlejohn))
- Fix wording in NamedandDefaultArguments.scala [\#187](https://github.com/scala-exercises/exercises-stdlib/pull/187) ([joelittlejohn](https://github.com/joelittlejohn))
- Fix typo, reverse not revert [\#184](https://github.com/scala-exercises/exercises-stdlib/pull/184) ([joelittlejohn](https://github.com/joelittlejohn))
- Explain that sameElements will work differently for a small set [\#183](https://github.com/scala-exercises/exercises-stdlib/pull/183) ([joelittlejohn](https://github.com/joelittlejohn))
- Minor formatting: the code part is 'Option', not 'Options' [\#182](https://github.com/scala-exercises/exercises-stdlib/pull/182) ([joelittlejohn](https://github.com/joelittlejohn))
- Fix typo: Stack not Stacks [\#178](https://github.com/scala-exercises/exercises-stdlib/pull/178) ([joelittlejohn](https://github.com/joelittlejohn))
- changed variable name [\#150](https://github.com/scala-exercises/exercises-stdlib/pull/150) ([rohanurkude28](https://github.com/rohanurkude28))

## [v0.6.2](https://github.com/scala-exercises/exercises-stdlib/tree/v0.6.2) (2020-04-27)

[Full Changelog](https://github.com/scala-exercises/exercises-stdlib/compare/v0.6.1...v0.6.2)

## [v0.6.1](https://github.com/scala-exercises/exercises-stdlib/tree/v0.6.1) (2020-04-27)

[Full Changelog](https://github.com/scala-exercises/exercises-stdlib/compare/v0.6.0...v0.6.1)

**Merged pull requests:**

- Compile when Publish [\#173](https://github.com/scala-exercises/exercises-stdlib/pull/173) ([juanpedromoreno](https://github.com/juanpedromoreno))

## [v0.6.0](https://github.com/scala-exercises/exercises-stdlib/tree/v0.6.0) (2020-04-24)

[Full Changelog](https://github.com/scala-exercises/exercises-stdlib/compare/v0.4.0...v0.6.0)

**Closed issues:**


project/ProjectPlugin.scala
import sbt.Keys._
import sbt._
import com.alejandrohdezma.sbt.github.SbtGithubPlugin

object ProjectPlugin extends AutoPlugin {

  override def trigger: PluginTrigger = allRequirements

  override def requires: Plugins = plugins.JvmPlugin && SbtGithubPlugin

  object autoImport {

    lazy val V = new {
      val shapeless: String           = "2.3.3"
      val scala: String               = "2.13.2"
      val scalaExercises: String      = "0.6.0"
      val scalacheck: String          = "1.14.3"
      val scalacheckShapeless: String = "1.2.5"
      val scalatest: String           = "3.1.2"
      val scalatestplusScheck: String = "3.1.1.1"
    }

    def dep(artifactId: String) = "org.scala-exercises" %% artifactId % V.scalaExercises

    lazy val exercisesSettings = Seq(
      libraryDependencies ++= Seq(
        dep("exercise-compiler"),
        dep("definitions"),
        "com.chuusai"                %% "shapeless"                 % V.shapeless,
        "org.scalatest"              %% "scalatest"                 % V.scalatest,
        "org.scalacheck"             %% "scalacheck"                % V.scalacheck,
        "org.scalatestplus"          %% "scalacheck-1-14"           % V.scalatestplusScheck,
        "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % V.scalacheckShapeless
      )
    )
  }

  override def projectSettings: Seq[Def.Setting[_]] =
    Seq(
      organization := "org.scala-exercises",
      organizationName := "47 Degrees",
      organizationHomepage := Some(url("https://47deg.com")),
      scalaVersion := autoImport.V.scala
    )
}
